Build failure: connman-gtk

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

```
nix-repl> :lf github:nixos/nixpkgs/a213bf831ea4791803dc4d0d01a812acdb09f126            
Added 16 variables.

nix-repl> :b legacyPackages.x86_64-linux.connman-gtk
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/286031301

### Relevant log output

```console
src/status.c: In function 'status_menu':
src/status.c:122:42: error: passing argument 1 of 'gtk_widget_set_sensitive' from incompatible pointer type [-Wincompatible-pointer-types]
  122 |                 gtk_widget_set_sensitive(item, has_items);
      |                                          ^~~~
      |                                          |
      |                                          GtkMenuItem * {aka struct _GtkMenuItem *}
In file included from /nix/store/5432m2b9nzij5g0lb2v8i6rj9g1hywdx-gtk+3-3.24.43-dev/include/gtk-3.0/gtk/gtkapplication.h:27,
                 from /nix/store/5432m2b9nzij5g0lb2v8i6rj9g1hywdx-gtk+3-3.24.43-dev/include/gtk-3.0/gtk/gtkwindow.h:33,
                 from /nix/store/5432m2b9nzij5g0lb2v8i6rj9g1hywdx-gtk+3-3.24.43-dev/include/gtk-3.0/gtk/gtkdialog.h:32,
                 from /nix/store/5432m2b9nzij5g0lb2v8i6rj9g1hywdx-gtk+3-3.24.43-dev/include/gtk-3.0/gtk/gtkaboutdialog.h:30,
                 from /nix/store/5432m2b9nzij5g0lb2v8i6rj9g1hywdx-gtk+3-3.24.43-dev/include/gtk-3.0/gtk/gtk.h:31,
                 from src/status.c:21:
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.2-zen1, NixOS, 25.05 (Warbler), 25.05.20250120.7ceed4f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/x46a0a227r8y8bqvld0d4hhxazlwsf3x-source`


### Notify maintainers

@romildo 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/co/connman-gtk/package.nix
+++ b/pkgs/by-name/co/connman-gtk/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch,
   autoconf,
   automake,
   intltool,
@@ -35,6 +36,13 @@ stdenv.mkDerivation rec {
     gtk3
     openconnect
     connman
+  ];
+
+  patches = [
+    (fetchpatch {
+      url = "https://salsa.debian.org/nickm/connman-gtk/-/raw/ef01b52fa02c5cca199b2e47c0cf360691266fd8/debian/patches/incompatible-pointer-type";
+      hash = "sha256-T+N9FfDyROBA4/HLK+l/fpnju2imDU4y6nGSbF+JDiA=";
+    })
   ];
 
   preConfigure = ''

